MobileWallet/Screens/Home/Transaction/TransactionViewController.swift

```diff
@@ -224,8 +234,138 @@ class TransactionViewController: UIViewController, UITextFieldDelegate {
+        guard let currentTx = transaction else {
+            return
+        }
+
+        guard let newTx = updatedTx else {
+            TariLogger.warn("Did not get transaction in callback reponse")
+            return
+        }
+
+        guard currentTx.id.0 == newTx.id.0 else {
+            //Received a tx update but it was for another tx
+            return
+        }
+
+        transaction = newTx
+
+        do {
+            try setDetails()
+            updateTxState()
+        } catch {
+            TariLogger.error("Failed to update TX state", error: error)
+        }
+    }
+
+    private func showStateView(defaultState: AnimatedRefreshingViewState, _ onComplete: @escaping () -> Void) {
+        guard !isShowingStateView else {
+            onComplete()
+            return
+        }
+
+        UIView.animate(withDuration: 0.5, delay: 0, options: .curveEaseInOut, animations: { [weak self] in
+            guard let self = self else { return }
+
+            self.navigationBarHeightAnchor.constant = self.defaultNavBarHeight + AnimatedRefreshingView.containerHeight + 10
+            self.navigationBar.layoutIfNeeded()
+            self.view.layoutIfNeeded()
+        }) { [weak self] (_) in
+            guard let self = self else { return }
+
+            self.txStateView.translatesAutoresizingMaskIntoConstraints = false
+            self.navigationBar.addSubview(self.txStateView)
+            self.txStateView.bottomAnchor.constraint(equalTo: self.navigationBar.bottomAnchor, constant: -Theme.shared.sizes.appSidePadding).isActive = true
+            self.txStateView.leadingAnchor.constraint(equalTo: self.navigationBar.leadingAnchor, constant: Theme.shared.sizes.appSidePadding).isActive = true
+            self.txStateView.trailingAnchor.constraint(equalTo: self.navigationBar.trailingAnchor, constant: -Theme.shared.sizes.appSidePadding).isActive = true
+
+            self.txStateView.setupView(defaultState)
+            self.txStateView.animateIn()
+
+            self.isShowingStateView = true
+            onComplete()
+        }
+    }
+
+    private func hideStateView() {
+        guard isShowingStateView else {
+            return
+        }
+
+        self.txStateView.animateOut { [weak self] in
+            guard let self = self else { return }
+
+            UIView.animate(withDuration: 0.5, delay: 0, options: .curveEaseInOut, animations: { [weak self] in
+                guard let self = self else { return }
+
+                self.navigationBarHeightAnchor.constant = self.defaultNavBarHeight
+                self.navigationBar.layoutIfNeeded()
+                self.view.layoutIfNeeded()
+            }) { [weak self] (_) in
+                guard let self = self else { return }
+                self.isShowingStateView = false
+            }
+        }
+    }
+
+    private func updateTxState() {
+        guard let tx = transaction else {
+            return
+        }
+
+        var newState: AnimatedRefreshingViewState?
```

Comment:
It got me a bit confused that the animated refresh view's state type is used in this class.